### PR TITLE
[cuda_xpu_alignment] F.gumbel_softmax(tau=0.0) silently produces all-NaN output instead of raising ValueError

### PR DESCRIPTION
**Pull request by Stonepia**
## [cuda_xpu_alignment] F.gumbel_softmax(tau=0.0) silently produces all-NaN output instead of raising ValueError

Fixes https://github.com/intel-sandbox/agentic_xpu/issues/15

**Root Cause:** torch.nn.functional.gumbel_softmax at torch/nn/functional.py:2229 computes `(logits + gumbels) / tau` without validating tau. When tau=0.0, division by zero produces inf/nan that propagates through softmax, yielding all-NaN output silently. Negative tau values are similarly unguarded despite the docstring implying tau should be positive.

---

**Diff stat:**
```
torch/nn/functional.py | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)
```

### CODE REVIEW

**Comment by Stonepia** [comment]
Closing: issue #15 (gumbel_softmax tau=0 NaN) is device-agnostic. Upstream pytorch/pytorch#184582 addresses this directly with a ValueError guard. No XPU-specific fix needed — this issue should be WAITING_UPSTREAM for #184582 to merge.
